core/node/consensus/src/storage/registry_contract/committee_extractor.rs

```diff
@@ -0,0 +1,136 @@
+use anyhow::Context;
+use zksync_concurrency::{ctx, ctx::Ctx, scope, time};
+use zksync_consensus_roles::{attester, validator};
+use zksync_dal::consensus_dal;
+use zksync_node_api_server::execution_sandbox::BlockArgs;
+
+use crate::storage::{
+    registry_contract::{abi::CommitteeAttester, vm_reader::VMReader},
+    ConnectionPool,
+};
+
+#[derive(Debug)]
+pub struct CommitteeExtractor {
+    pool: ConnectionPool,
+    reader: VMReader,
+}
+
+impl CommitteeExtractor {
+    pub fn new(pool: ConnectionPool, reader: VMReader) -> CommitteeExtractor {
+        Self { pool, reader }
+    }
+
+    pub async fn run(mut self, ctx: &Ctx) -> anyhow::Result<()> {
+        tracing::info!("Running zksync_node_consensus::storage::CommitteeExtractor");
+        let res = scope::run!(ctx, |ctx, s| async {
+            let mut next_batch = self.next_batch_to_extract_committee(ctx).await?;
+            loop {
```

Comment:
Would it be possible at all to hook into the batch execution itself, so instead of polling and eventually loading the data from here to there, the derived data would be inserted within the same transaction?